lib/helpers/validator.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+var pkg = require('./../../package.json');
+
+var validators = {};
+
+// eslint-disable-next-line func-names
+['object', 'boolean', 'number', 'function', 'string', 'symbol'].forEach(function(type, i) {
+  validators[type] = function validator(thing) {
+    return typeof thing === type || 'a' + (i < 1 ? 'n ' : ' ') + type;
+  };
+});
+
+var deprecatedWarnings = {};
+var currentVerArr = pkg.version.split('.');
```

Comment:
`var pkg = require('./../../package.json');` tries to load my project's package.json which has no version key set, causing the application to fail on boot in the browser. 
what's the purpose of this validator and why does it need my project's package.json?
it's a vue3/webpack project.

Reply:
Will have a look as to why this is happening

Reply:
it's an edge cas, really - people do the weirdest things in package.json (as my own exampled showed). the fix is simple enough, the package.json just has to be valid and meet the minimum requirements.
yet, this issue might highlight another issue as relying and bundling the package.json might not be the best solution to use a version for negotiation (as seen in this PR) nor to be used for the user agent (as seen in the linked issue above where it is imported in the http adapter).

the easiest way to make it more robust is to skip this PRs version parsing if none is found (even though it should be there, technically - til).
or this is actually just a path thing and the version to be read is actually the one from axios' own package.json, not the project's?

Reply:
I meet the same question here, so do you know how to solve it?

Reply:
> I meet the same question here, so do you know how to solve it?

For me the solution was to add a version to `package.json`. `"version": "0.0.0"` works just fine.